src/index.ts

```diff
@@ -1,132 +1,231 @@
+    const buf = newWasmBuffer({ data: binToString(bytes) });
+    const result = call("wasm_decode_call", buf);
+    const json = loadWasmBufferDataAsJson(result);
+    wasmBufferFree(buf);
+    wasmBufferFree(result);
+    return json;
+}
+
+// Encodes data provided in object and returns the encoded data in a json object
+export function encodeInput(object: any) : JSON {
+    const buf = newWasmBuffer(object);
+    const result = call("wasm_encode_inputdata", buf);
+    const encoded = loadWasmBufferDataAsJson(result);
+    wasmBufferFree(buf);
+    wasmBufferFree(result);
+    return encoded;
+}
+
+// Decode SVM data provided in encodedData value and returns a json object of the data
+export function decodeInput(encodedData: any) : JSON {
+    const buf = newWasmBuffer(encodedData);
+    const result = call("wasm_decode_inputdata", buf);
+    const json = loadWasmBufferDataAsJson(result);
+    wasmBufferFree(buf);
+    wasmBufferFree(result);
+    return json;
+}
+
+
+// Frees an allocated svm_codec buffer that was previously allocated and returned to caller by an api function
+export function wasmBufferFree(buf) : void {
+    return (codec.exports as any).wasm_free(buf);
+}
+
+// internal helper functions
+
+// Encodes binary provided binary data as a hex binary string (without an 0x prefix)
+function binToString(array: Uint8Array) : string {
+    let result = "";
+    for (const b of array) {
+        let s = b.toString(16);
+
+        // padding
+        if (s.length < 2) {
+            s = "0" + s;
+        }
+        result += s;
+    }
+    return result;
+}
+
+// Call an svm_codec function with the provided buffer. Returns result buffer.
+function call(funcName: string, buf) : Uint8Array {
+    return (codec.exports as any)[funcName](buf)
 }
 
-// Frees an allocated svm_codec buffer that was previously allocated by svm_codec
-export function wasmBufferFree(buf) {
-    return (codec!.exports as any).wasm_free(buf);
+// Allocates a svm_codec buffer with the provided byte length
+function wasmBufferAlloc(length: number) {
+    return (codec.exports as any).wasm_alloc(length)
 }
 
 // Returns the bytes length of a wasm_codec buffer
-export function wasmBufferLength(buf) {
-    return (codec!.exports as any).wasm_buffer_length(buf);
+function wasmBufferLength(buf) {
+    return (codec.exports as any).wasm_buffer_length(buf);
 }
 
 // Frees the data allocated in a svm_codec buffer
-export function wasmBufferDataPtr(buf) {
-    return (codec!.exports as any).wasm_buffer_data(buf);
+function wasmBufferDataPtr(buf) {
+    return (codec.exports as any).wasm_buffer_data(buf);
 }
 
 // Copies binary data from buf to an svm_codec memory buffer
-export function copyToWasmBufferData(buf, data) {
-    let ptr = wasmBufferDataPtr(buf);
-    let memory = (codec!.exports as any).memory.buffer;
-    let view = new Uint8Array(memory);
+function copyToWasmBufferData(buf, data) {
+    const ptr = wasmBufferDataPtr(buf);
```

Comment:
@YaronWittenstein

Reply:
Seems that we don't really need it, since `copyToWasmBufferData` is an internal function.
But, it's still a good practice to add an assertion.

So @avive the `svm_codec.wasm` exposes `wasm_buffer_length`:
https://github.com/spacemeshos/svm/blob/d91e8e76f81aa9319a4c4f109a6a9372ca0065c5/crates/codec/src/api/wasm/mod.rs#L81

You can grab that and assert that the length of `data` fits into the `Data`-length pointed by SVM Wasm Buffer `buf`

Reply:
I' like to have these kind improvements suggestions in follow up issues and prs. I will create them.